### ISSUE TITLE
Arrival day

### ISSUE THREAD

**Issue by strarsis** [wakirin/Litepicker#97]
**Is your feature request related to a problem? Please describe.**
For same date ranges the arrival day should be enforced. The user shouldn't select any other days of the week than those allowed (e.g. only Mondays), but can still freely select the departure date (e.g. being a Friday).

The `bookedDays` looks like as it does exactly what I described. However, it is not possible to select a start date in front of a "booked day" and then select an end date after the "booked day" as it should be possible, as that option is described to lock the start selection, not the end selection and the `disallowLockDaysInRange` option is for `lockedDays`, not `bookedDays`!
You can reproduce this issue in this pen: https://codepen.io/strarsis/pen/ZEQzreb
One booked day is set up, 10.06.2020. It is not possible to select a start day before it and and an end day after it.

**Describe the solution you'd like**
A more flexible mechanism like a callback for allowing the selectability of days in calendar or a dedicated option (which belongs to hotel mode).

**Describe alternatives you've considered**
Disallowing those dates, though this is also a recurring day in the week, so this can be quite wasteful/complicated. Another issue is that those disallowed days of week look like unavailable days of the week, which is confusing. It would be better if those days that are not allowed as a day of the week are a bit greyed out, but not just locked.

**Additional context**
Basically I want to select + style (and change state (selectable to unselectable)) of each calendar item, some functions, and a callback would offer the most flexible way doing so.

**Comment by strarsis** [comment]
Addendum: Being able to specify a date/time period and then apply rules, e.g. "locked" or custom classes to the days/blocks that match would make all the things like arrival days possible.

**Comment by wakirin** [comment]
possibly related to #72
